### ISSUE TITLE
Consumer using significantly more network bandwidth than the size of consumed messages

### ISSUE THREAD

**Issue by jeremydonahue** [edenhill/librdkafka#1504]
Description
===========

_librdkafka_ is using significantly more network bandwidth than the size of consumed messages.

**Bandwidth used per second via `iftop`**:
![iftop](https://user-images.githubusercontent.com/14008978/32400821-60afcd88-c0c2-11e7-8648-2670f500eeb8.png)
![iftop-2](https://user-images.githubusercontent.com/14008978/32400895-8d7bbf92-c0c3-11e7-8a95-e03f8aea4e5f.png)

**Size of messages consumed**:
```
$ ls -alh output
-rw-r--r-- 1 root root  87M Nov  3 18:06 output
```
Based on the network throughput (4.65Gbps), and a runtime of `real    1m25.349s`, I would expect to see 49.3Gbytes of data consumed. There are no errors produced by _librdkafka_ during these tests. Am I doing something wrong?

Thanks!

How to reproduce
================

I have been able to reproduce this using 3 different bindings to _librdkafka_. The input data is ~1kb/line and was written to a single topic with 40 partitions (5 brokers), `batch.num.messages: 10000`, and `snappy` compression.

- [kafkacat](https://github.com/edenhill/kafkacat)
```$ ./kafkacat -C -t topic -b broker >> output```

- [confluent-kafka-go/consumer_channel_example.go](https://github.com/confluentinc/confluent-kafka-go/blob/master/examples/consumer_channel_example/consumer_channel_example.go)
```$ ./consumer_channel_example broker:9092 group topic >> output```

- [confluent-kafka-python](https://github.com/confluentinc/confluent-kafka-python)
```
import sys 
from confluent_kafka import Consumer, KafkaError

config = {"bootstrap.servers":    sys.argv[1],
          "group.id":             sys.argv[2],
          "session.timeout.ms":   6000,
          "default.topic.config": {"auto.offset.reset": "earliest"}}

c = Consumer(**config)
c.subscribe([sys.argv[3]])

running = True
while running:
    msg = c.poll()
    if not msg.error():
        print(msg.value())
    elif msg.error().code() != KafkaError._PARTITION_EOF:
        print(msg.error())
        running = False
c.close()
```
```$ python script.py >> output```

Checklist
=========
Please provide the following information:

 - [x] librdkafka version (release number or git tag): 0.11.0 (Confluent-platform 3.3.0)
 - [x] Apache Kafka version: 0.11.0
 - [x] librdkafka client configuration: see above
 - [x] Operating system: Ubuntu Trusty (kernel version: 4.4.0-62-generic)
 - [ ] Using the legacy Consumer
 - [x] Using the high-level KafkaConsumer
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by edenhill** [comment]
This is a great bug report!

Good news is that the issue was fixed in v0.11.1:
https://github.com/edenhill/librdkafka/issues/1384

**Comment by jeremydonahue** [comment]
I confirmed this problem is fixed with v0.11.1+. Thanks for the response!

**Comment by edenhill** [comment]
Thanks! :+1:
